src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.html

```diff
@@ -1,3 +1,153 @@
+                       (change)="toggleAllPresentTeamDetailsState(!isAllPresentTeamDetailsChecked)"
+                       [checked]="isAllPresentTeamDetailsChecked">
+                <label class="ml-1"><strong> Select All </strong></label>
+              </div>
+              <ng-container *ngFor="let courseDetails of courseDetailsList">
+                <ng-container *ngFor="let section of courseDetails.sections">
+                  <div class="checkbox-outer-div" *ngFor="let team of section.teams" [hidden]="!section.isChecked" >
+                    <input type="checkbox" [checked]="team.isChecked" (change)="toggleTeamState(team)">
+                    <label class="ml-1">
+                      [{{courseDetails.id}}]: {{team.name}}
+                    </label>
+                  </div>
+                </ng-container>
+              </ng-container>
+            </ng-container>
+          </div>
+
+          <div class="col-sm-3 float-left">
+            <ng-container *ngIf="courseDetailsList.length">
+              <div class="text-color-primary pb-3">
+                <strong> Students </strong>
+              </div>
+              <div class="checkbox-outer-div pb-3" [hidden]="!isAnyPresentCoursesChecked">
+                <input type="checkbox"
+                       (change)="toggleAllPresentStudentDetailsState(!isAllPresentStudentDetailsChecked)"
+                       [checked]="isAllPresentStudentDetailsChecked">
+                <label class="ml-1"><strong> Show Emails </strong></label>
+              </div>
+              <ng-container *ngFor="let courseDetails of courseDetailsList">
+                <ng-container *ngFor="let section of courseDetails.sections">
+                  <ng-container *ngFor="let team of section.teams">
+                    <div class="checkbox-outer-div" *ngFor="let student of team.students" [hidden]="!team.isChecked">
+                      <input type="checkbox" [(ngModel)]="student.isChecked">
+                      <label class="ml-1">
+                        {{student.email}}
+                      </label>
+                    </div>
+                  </ng-container>
+                </ng-container>
+              </ng-container>
+            </ng-container>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+<ng-container *ngIf="courses && courses.length" >
+  <div class="p-2">
+    <div *ngFor="let course of courses" class="card border mb-3 p-0"
+         [hidden]="(!course.isChecked) && (!isDisplayArchive) && course.isArchived">
+      <div class="card-header font-weight-bold"
+           [ngClass]="{'text-dark-blue bg-light-blue ': !course.isArchived, 'bg-light-grey': course.isArchived}"
+           (click)="toggleCourseStateAtInput(course)">
+        <strong>[{{course.id}}]:</strong> {{course.name}}
+        <div class="arrow-sign">
+          <button class="btn btn-info btn-xs" (click)="enroll(course.id)"> Enroll </button>
```

Comment:
Do you really need this (`enroll`) to be a function? Judging by the usage you can just pass `routerLink` to this button element.